lib/popcorn/pop_tracker.nit

```diff
@@ -0,0 +1,238 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Popcorn web tracker
+#
+# Easy and ready to use web tracker you can apply to your popcorn application.
+#
+# The only thing you have to do is to use the tracker in your app routes:
+
+# ~~~nitish
+# var config = new AppConfig
+# var app = new App
+# app.use("/", new HomeHandler)
+# app.use("/products", new ProductsHandler)
+# app.use("customers/", new CustomersHandler)
+#
+# app.use_after("/*", new PopTracker(config)) # tracker listens to /*
+# ~~~
+#
+# You can also use multiple tracker at once on different route.
+# All the data will be aggregated for you.
+
+# ~~~nitish
+# app.use_after("/api/*", new PopTracker(config))
+# app.use_after("/admin/*", new PopTracker(config))
+# ~~~
+#
+# To retrieve your tracker data use the `PopTrackerAPI` which serves the tracker
+# data in JSON format.
+#
+# ~~~nitish
+# app.use("/api/tracker_data", new PopTrackerAPI(config))
+# ~~~
+module pop_tracker
+
+import popcorn
+import popcorn::pop_config
+import popcorn::pop_logging
+import popcorn::pop_repos
+
+redef class AppConfig
+
+	# Logs collection
+	var tracker_logs = new TrackerRepo(db.collection("tracker_logs"))
+
+	# Tracker handler
+	var tracker = new PopTracker(self)
+end
+
+# JSON API of the PopTracker
+#
+# Serves the collected data in JSON format.
```

Comment:
The API should be documented so the users don’t have to guess how to use it (or dig in the source code).